lib/iam.js

```diff
@@ -0,0 +1,14 @@
+const AWS = require('aws-sdk')
+
+function decorateWithIamToken(baseConfig) {
+    const rdsSigner = new AWS.RDS.Signer()
+    const token = rdsSigner.getAuthToken({
```

Comment:
does `getAuthToken(..)` return a promise?

If so, maybe make this function async (e.g. `async function decorateWithIamToken`..).
and `await rdsSigner.getAuthToken` ?

Reply:
No, that function is synchronous in the aws-sdk, it also has an async variant which takes a callback argument but that is not the one i'm using in the prod-code.

I think that aws-sdk-mocking library probably can't distinguish between the two properly, resulting in this awkward test-implementation. It's that or I'm using the mocking-lib incorrectly.

Reply:
On second (or third) thought, apparently the `CredentialsProvider` from the `CredentialProviderChain` that gets selected by `RDS.Signer` in the context of a Lambda is sync. This is why I see my code working and producing files into the s3 bucket nicely. So, I don't think it is required to refactor this code so that it provides a `callback` argument to.